.github/workflows/regression-report.yml

```diff
@@ -81,30 +99,177 @@ jobs:
+          PROJECTS_LINK="$BASE_URL/$CONFIG_BUNDLE/list-of-projects.properties"
           ./.ci/append-to-github-output.sh "config_link" "$CONFIG_LINK"
           ./.ci/append-to-github-output.sh "projects_link" "$PROJECTS_LINK"
 
-      - name: Set branch
+  handle_generate_config:
+    needs: parse_comment
+    if: needs.parse_comment.outputs.mode == 'generate_config'
+    runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.generate_config.outputs.config_link }}
+      projects_link: ${{ steps.generate_config.outputs.projects_link }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Checkout Checkstyle repository
+        uses: actions/checkout@v4
+        with:
+          repository: checkstyle/checkstyle
+          path: .ci-temp/checkstyle
+
+      - name: Find input file
+        id: find_input_file
+        run: |
+          INPUT_FILE="${{ needs.parse_comment.outputs.input_file }}"
+          CHECKSTYLE_DIR=".ci-temp/checkstyle"
+          FILENAME=$(basename "$INPUT_FILE")
+
+          # Remove .java extension if present
+          FILENAME_NO_EXT="${FILENAME%.java}"
+
+          FOUND_FILE=$(find "$CHECKSTYLE_DIR" -name "${FILENAME_NO_EXT}*.java" -type f)
+          if [ -z "$FOUND_FILE" ]; then
+            echo "Error: Could not find input file"
+            exit 1
+          fi
+          ./.ci/append-to-github-output.sh "input_file" "$FOUND_FILE"
+
+      - name: Download DiffTool JAR
+        uses: robinraju/release-downloader@v1.8
+        with:
+          repository: "relentless-pursuit/test-configs"
+          tag: "extractor-1.0.0"
+          fileName: "extractor-all.jar"
+
+      - name: Set up Java 21
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Generate config
+        id: generate_config
         env:
+          DEFAULT_PROJECTS_LINK: ${{ env.DEFAULT_PROJECTS_LINK }}
+        run: |
+          BASE_URL="https://raw.githubusercontent.com/checkstyle/test-configs/main"
+          FOUND_FILE="${{ steps.find_input_file.outputs.input_file }}"
+          BASE_PATH=".ci-temp/checkstyle"
+          OUTPUT_DIR="generated_configs"
+          mkdir -p "$OUTPUT_DIR"
+          
+          echo "Debug: Original FOUND_FILE = $FOUND_FILE"
+          
+          # Remove .ci-temp/checkstyle/ from the beginning of FOUND_FILE
+          RELATIVE_PATH=${FOUND_FILE#.ci-temp/checkstyle/}
+          
+          echo "Debug: Processed RELATIVE_PATH = $RELATIVE_PATH"
+          
+          # Generate a unique name for the config file
+          CONFIG_FILE_NAME=$(basename "$FOUND_FILE" .java)_config.xml
+          CONFIG_FILE_PATH="$OUTPUT_DIR/$CONFIG_FILE_NAME"
+          
+          echo "Debug: CONFIG_FILE_PATH = $CONFIG_FILE_PATH"
+          
+          # Run the extractor
+          java -jar extractor-all.jar "$BASE_PATH" --input-file "$RELATIVE_PATH" "$CONFIG_FILE_PATH"
+
```

Comment:
Please print by cat content of file, to see it in logs in case there's some problems

Reply:
done